From 1be5f0c6f0e9bfa1181ce47631be64acdb1934c0 Mon Sep 17 00:00:00 2001
From: Peter Hutterer <peter.hutterer@who-t.net>
Date: Wed, 25 Aug 2010 14:55:48 +1000
Subject: [PATCH 4/9] Remove superfluous checks for TVResolution property values.

A CARD32 is unsigned and cannot be less than 0. Remove the checks and move
the one leftover check down to the switch statement.
---
 src/wcmXCommand.c |    7 ++-----
 1 files changed, 2 insertions(+), 5 deletions(-)

diff --git a/src/wcmXCommand.c b/src/wcmXCommand.c
index 4c81b9c..4716840 100644
--- a/src/wcmXCommand.c
+++ b/src/wcmXCommand.c
@@ -549,13 +549,10 @@ int wcmSetProperty(DeviceIntPtr dev, Atom property, XIPropertyValuePtr prop,
 		values = (CARD32*)prop->data;
 
 		/* non-TwinView settings can not set TwinView RESOLUTION */
-		if ((priv->twinview == TV_NONE) || (values[0] < 0) ||
-				(values[1] < 0) || (values[2] < 0) || (values[3] < 0))
-			return BadValue;
-
 		switch(priv->twinview)
 		{
-			case TV_NONE: break;
+			case TV_NONE:
+				return BadValue;
 			case TV_ABOVE_BELOW:
 			case TV_BELOW_ABOVE:
 				      if ((values[1] + values[3]) != screenInfo.screens[0]->height)
-- 
1.7.2.2

